src/store/user-context.tsx

```diff
@@ -1,43 +1,88 @@
-import React, { useState, useReducer } from "react";
+import React, { useReducer } from "react";
+import {
+  Props,
+  UserContextObject,
+  CartItemObject,
+} from "../common/types/common.types";
 
-type UserContextObject = {
-  order: {
-    items: Object[];
-    adress: {};
-  };
-  history: [];
-  sortType: string;
-  addItem: () => void;
-  removeItem: () => void;
-};
-
-type Props = { children: React.ReactNode };
-
-// default: empty values
-export const UserContext = React.createContext<UserContextObject>({
-  order: {
-    items: [],
-    adress: {},
-  },
+// Creating default context state object
+const defaultUserState: UserContextObject = {
+  cartItems: [],
+  address: {},
   history: [],
   sortType: "",
   addItem: () => {},
   removeItem: () => {},
-});
+  clearCart: () => {},
+};
+
+// Creating context
+export const UserContext = React.createContext<UserContextObject>(
+  Object.assign(defaultUserState)
+);
+
+// Creating reducer function
+const userReducer = (
+  state: UserContextObject,
+  action: { type: string; item?: CartItemObject; id?: string }
+) => {
+  if (action.type === "ADD_ITEM") {
```

Comment:
it would be better to create a file with constants e.g.
`
export enum CartActionTypes {
  ADD_ITEM = 'ADD_ITEM',
  REMOVE_ITEM = 'REMOVE_ITEM',
  CLEAR_CART = 'CLEAR_CART',
}
`